src/plugins/vis_builder/public/application/utils/use/use_saved_vis_builder_vis.ts

```diff
@@ -73,11 +74,13 @@ export const useSavedVisBuilderVis = (visualizationIdFromUrl: string | undefined
 
           const validateResult = validateVisBuilderState({ styleState, visualizationState });
           if (!validateResult.valid) {
-            const err = validateResult.errors;
-            if (err) {
-              const errMsg = err[0].instancePath + ' ' + err[0].message;
-              throw new InvalidJSONProperty(errMsg);
-            }
+            throw new InvalidJSONProperty(
+              validateResult.errorMsg ||
+                i18n.translate('visBuilder.useSavedVisBuilderVis.genericJSONError', {
+                  defaultMessage:
+                    'Something went wrong while loading your saved object. The object may be corrupted or does not match the latest schema',
```

Comment:
This will likely never be show to the user because the ajv validator responsible for validating the schema will always have an error message associated with an invalid result. This was just me not wanting override typescripts `undefined` check because this prop is marked optional. In future if the function ever changes and the error message is actually empty, we have a generic error as opposed to an error. Atleast that was my thinking.